wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/validator/DevfileSchemaValidator.java

```diff
@@ -55,14 +55,41 @@ public JsonNode validateBySchema(String yamlContent, boolean verbose)
       throw new DevfileFormatException("Unable to validate Devfile. Error: " + e.getMessage());
     }
     if (!report.isSuccess()) {
-      String error =
-          StreamSupport.stream(report.spliterator(), false)
-              .filter(m -> m.getLogLevel() == LogLevel.ERROR || m.getLogLevel() == LogLevel.FATAL)
-              .map(message -> verbose ? message.asJson().toString() : message.getMessage())
-              .collect(Collectors.joining(", ", "[", "]"));
-      throw new DevfileFormatException(
-          format("Devfile schema validation failed. Errors: %s", error));
+      throw new DevfileFormatException(prepareErrorMessage(report));
     }
     return data;
   }
+
+  private String prepareErrorMessage(ProcessingReport report) {
```

Comment:
Could you please move it in a separate class, add detailed java doc, and some unit tests